vulture/core.py

```diff
@@ -205,14 +236,23 @@ def get_set(typ):
 
         self.ignore_names = ignore_names or []
         self.ignore_decorators = ignore_decorators or []
+        self.noqa_matches = defaultdict(set)
 
         self.filename = ""
         self.code = []
         self.found_dead_code_or_error = False
 
+    def parse_noqa(self, code):
```

Comment:
It's a bit strange that the method is passed `self.code`, but not `self.noqa_matches` even though both are instance attributes. I think we should pass both variables to the method and turn it into a function. Then we can collect all code related to noqa in a vulture/noqa.py file, i.e., parse_noqa(), _parse_error_codes() and NOQA_REGEXP.